### ISSUE TITLE
jay doesn't create its own cache directory

### ISSUE THREAD

**Issue by peterc** [nikersify/jay#7]
Installing Jay on a fresh install and then running it yields:

```
Error: ENOENT: no such file or directory, open '/Users/peter/Library/Caches/jay-repl-nodejs/history'
    at Object.openSync (fs.js:447:3)
    at Object.createHistorian (/usr/local/lib/node_modules/jay-repl/dist/history.js:12:41)
```

`mkdir /Users/peter/Library/Caches/jay-repl-nodejs` resolved the situation, but I assume this should be happening without my intervention.

**Comment by superhawk610** [comment]
This should be as simple as ensuring the directory exists before attempting to touch history [here](https://github.com/nikersify/jay/blob/master/source/history.ts#L8).

```ts
export function createHistorian(filePath: string) {
	ensureDir(path.dirname(filePath))
	fs.closeSync(fs.openSync(filePath, 'a')) // Touch the history file
	// ...
}

function ensureDir(path: string) {
	if (!fs.existsSync(path)) {
		fs.mkdirSync(path, { recursive: true })
	}
}
```

You can get a more robust and backwards-compatible `ensureDir` from https://www.npmjs.com/package/fs-extra.

**Comment by nikersify** [comment]
fixed in 0.2.1, thanks for the report!
